src/System.Private.CoreLib/shared/Interop/Windows/Advapi32/Interop.EventRegister.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class Advapi32
+    {
+        [DllImport(Libraries.Advapi32, ExactSpelling = true)]
+        internal static extern unsafe uint EventRegister(
+            [In] ref Guid providerId,
+            [In] EtwEnableCallback enableCallback,
+            [In] void* callbackContext,
```

Comment:
Most of these `[In]` attributes are superfluous.

Reply:
I know. I kept it from the original code, but I can clean it up.